app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+                    SynchronizationProviderViewData.getUsernameFromSelectedSyncProvider(
+                            getContext(),
+                            getSelectedSyncProviderKey()
+                    ),
+                    getActivity().getString(R.string.pref_synchronization_logout_summary));
+            Spanned formattedSummary = HtmlCompat.fromHtml(summary, HtmlCompat.FROM_HTML_MODE_LEGACY);
+            findPreference(PREFERENCE_LOGOUT).setSummary(formattedSummary);
+            updateLastSyncReport(SyncService.isLastSyncSuccessful(getContext()),
+                    SyncService.getLastSyncAttempt(getContext()));
+        } else {
+            findPreference(PREFERENCE_LOGOUT).setSummary(null);
+        }
+    }
+
+
+
+    private int getIconForSelectedSyncProvider(String currentSyncProviderKey) {
+        return SynchronizationProviderViewData.getSynchronizationProviderIcon(getContext(), currentSyncProviderKey);
+    }
+
+    private int getHeaderSummary(String currentSyncProviderKey) {
+        return SynchronizationProviderViewData.getSynchronizationProviderHeaderSummary(
+                getContext(),
+                currentSyncProviderKey
+        );
+    }
+
+    private boolean isGpodnetSyncProviderSelected() {
+        return getSelectedSyncProviderKey().equals(SYNC_PROVIDER_CHOICE_GPODDER_NET);
+    }
+
+    private String getSelectedSyncProviderKey() {
+        return getActivity()
+                .getSharedPreferences(SyncService.SHARED_PREFERENCES_SYNCHRONIZATION, Activity.MODE_PRIVATE)
+                .getString(SyncService.SHARED_PREFERENCE_SELECTED_SYNC_PROVIDER, SYNC_PROVIDER_UNSET);
+    }
+
+    private void updateLastSyncReport(boolean successful, long lastTime) {
+        String status = String.format("%1$s (%2$s)", getString(successful
+                        ? R.string.gpodnetsync_pref_report_successful : R.string.gpodnetsync_pref_report_failed),
+                DateUtils.getRelativeDateTimeString(getContext(),
+                        lastTime, DateUtils.MINUTE_IN_MILLIS, DateUtils.WEEK_IN_MILLIS, DateUtils.FORMAT_SHOW_TIME));
+        ((PreferenceActivity) getActivity()).getSupportActionBar().setSubtitle(status);
+    }
+
+    private void openNextcloudAccountChooser() {
+        try {
+            AccountImporter.pickNewAccount(this);
+        } catch (NextcloudFilesAppNotInstalledException | AndroidGetAccountsPermissionNotGranted e) {
+            UiExceptionManager.showDialogForException(getContext(), e);
+        }
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        try {
+            AccountImporter.onActivityResult(
+                    requestCode,
+                    resultCode,
+                    data,
+                    SynchronizationPreferencesFragment.this,
+                    singleSignOnAccount
+                            -> {
```

Comment:
This is taking up a lot of vertical screen space. Please write arguments in one line and only add a line break when the line needs more than 120 characters.
```suggestion
            AccountImporter.onActivityResult(requestCode, resultCode, data, SynchronizationPreferencesFragment.this,
                    singleSignOnAccount -> {
```